html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/OAuth/Instagram.pm

```diff
@@ -0,0 +1,178 @@
+package captiveportal::PacketFence::DynamicRouting::Module::Authentication::OAuth::Instagram;
+
+=head1 NAME
+
+captiveportal::DynamicRouting::Module::Authentication::OAuth::Instagram
+
+=head1 DESCRIPTION
+
+Instagram OAuth module
+
+=cut
+
+use WWW::Curl::Easy;
+use Moose;
+use pf::log;
+extends 'captiveportal::DynamicRouting::Module::Authentication::OAuth';
+
+has '+source' => (isa => 'pf::Authentication::Source::InstagramSource');
+
+=head2 get_token
+
+Get the OAuth2 token
+
+=cut
+
+sub get_token {
+    my ($self) = @_;
+
+    my $curl = WWW::Curl::Easy->new;
+    my $code = $self->app->request->parameters->{code};
+    my $info = $self->get_client;
+
+    my $source = $self->source;
+
+    my $response_body = '';
+    my $formdata = WWW::Curl::Form->new;
+    
+    $formdata->formadd("client_id", "$info->{NOP_id}");
```

Comment:
remove surrounding quotes in `"$info->{NOP_id}"` and do the same for client_secret and redirect_uri